scripts/histmakers/w_z_gen_dists.py

```diff
@@ -246,10 +250,10 @@ def build_graph(df, dataset):
     if z_moments:
         z_moments = hh.rebinHist(z_moments, axis_ptVgen.name, common.ptV_binning)
         z_moments = hh.rebinHist(z_moments, axis_massZgen.name, axis_massZgen.edges[::2])
-        coeffs["Z"] = wremnants.moments_to_angular_coeffs(z_moments)
+        coeffs["Z"] = wremnants.moments_to_angular_coeffs(z_moments, sumW2=True) # sumW2=True depends on how histograms are defined before, must be consistent
```

Comment:
So here when I changed some histograms to have storage weight (from original Double) I got a crash. Something happens within narf or boost that requires to read content as value, variance pairs, but is sumW2 is kept to False it assumes it has a single value. Otherwise I have to revert the storage to what it originally was (but Tanmay was telling me he needs the storage weights)

Reply:
Well sumWeights=False should be fixed in this function, then. There's no reason it shouldn't work

Reply:
I think it is because that function internally sets:
sumW2 = sumW2 and hist_moments_scales._storage_type == hist.storage.Weight

so even when you have weights you need to set sumW2 explicitly to True. Is there a reason it was like that? I would guess sumW2 = hist_moments_scales._storage_type == hist.storage.Weight should be sufficient

Reply:
I could propose to use the logical OR (so when the histogram has weights you should definitely use sumW2 = True, otherwise if it is double you may still chose to treat it as having weights but set them to 0.